### PR TITLE
make-squashfs,make-iso9660-image: use `__structuredAttrs`

### PR DESCRIPTION
**Pull request by ncfavier**
Makes it easier to enable [discarding of references](https://github.com/NixOS/nix/pull/7087), an upcoming feature of Nix which requires structured attrs.

This also makes the builder for make-iso9660-image more robust since we can pass proper arrays instead of space-separated strings.

For reference, `__structuredAttrs` is supported since Nix 2.0 and nixpkgs https://github.com/NixOS/nixpkgs/pull/175649.

### CODE REVIEW

File: nixos/lib/make-iso9660-image.nix [hunk 1]
```diff
@@ -47,16 +47,18 @@ assert usbBootable -> isohybridMbrImage != "";
...
-  builder = ./make-iso9660-image.sh;
+  __structuredAttrs = true;
+
+  buildCommand = builtins.readFile ./make-iso9660-image.sh;
   nativeBuildInputs = [ xorriso syslinux zstd libossp_uuid ];
 
-  inherit isoName bootable bootImage compressImage volumeID efiBootImage efiBootable isohybridMbrImage usbBootable;
+  env = {
+    inherit isoName bootable bootImage compressImage volumeID efiBootImage efiBootable isohybridMbrImage usbBootable;
+  };
```
Comment (alyssais) [inline nixos/lib/make-iso9660-image.nix:57]: Most/all of these probably don't need to be exported, right?

File: nixos/lib/make-iso9660-image.nix [hunk 2]
```diff
@@ -47,16 +47,16 @@ assert usbBootable -> isohybridMbrImage != "";
 
 stdenv.mkDerivation {
   name = isoName;
-  builder = ./make-iso9660-image.sh;
+  __structuredAttrs = true;
+
+  buildCommand = builtins.readFile ./make-iso9660-image.sh;
```
Comment (alyssais) [inline nixos/lib/make-iso9660-image.nix:52]: Can we use `buildCommandPath`?

Comment (ncfavier) [reply nixos/lib/make-iso9660-image.nix]: We can

**Review by alyssais — APPROVED** [review]

**Comment by cole-h** [comment]
@ofborg eval

**Pull request merged**